frontend/src/canvas/Canvas.js

```diff
@@ -1,50 +1,81 @@
-import React, { useCallback, useRef, useEffect, useState } from 'react'
-import useWebSocket, { ReadyState } from 'react-use-websocket'
-import './Canvas.css';
+import React, { useCallback, useRef, useEffect, useState } from "react";
+import useWebSocket, { ReadyState } from "react-use-websocket";
+import "./Canvas.css";
 // import TemplateOverlay from './TemplateOverlay.js';
-import canvasConfig from "../configs/canvas.config.json"
-import backendConfig from "../configs/backend.config.json"
+import canvasConfig from "../configs/canvas.config.json";
+import backendConfig from "../configs/backend.config.json";
 
-const Canvas = props => {
-  const backendUrl = "http://" + backendConfig.host + ":" + backendConfig.port
+const Canvas = (props) => {
+  const backendUrl = "http://" + backendConfig.host + ":" + backendConfig.port;
   // TODO: Pressing "Canvas" resets the view / positioning
 
-  const [canvasPositionX, setCanvasPositionX] = useState(0)
-  const [canvasPositionY, setCanvasPositionY] = useState(0)
-  const [isDragging, setIsDragging] = useState(false)
-  const [dragStartX, setDragStartX] = useState(0)
-  const [dragStartY, setDragStartY] = useState(0)
+  const [canvasPositionX, setCanvasPositionX] = useState(0);
+  const [canvasPositionY, setCanvasPositionY] = useState(0);
+  const [isDragging, setIsDragging] = useState(false);
+  const [dragStartX, setDragStartX] = useState(0);
+  const [dragStartY, setDragStartY] = useState(0);
 
-  const [canvasScale, setCanvasScale] = useState(6)
-  const minScale = 1 // TODO: To config
-  const maxScale = 40
+  const [canvasScale, setCanvasScale] = useState(6);
+  const minScale = 1; // TODO: To config
+  const maxScale = 40;
   //TODO: Way to configure tick rates to give smooth xp for all users
-  
-  const canvasRef = useRef(null)
+
+  const canvasRef = useRef(null);
 
   // Read canvas config from environment variable file json
-  const width = canvasConfig.canvas.width
-  const height = canvasConfig.canvas.height
-  const colors = canvasConfig.colors
+  const width = canvasConfig.canvas.width;
+  const height = canvasConfig.canvas.height;
+  // const colors = canvasConfig.colors
+
+  const [colors, setColors] = useState({});
+
+  useEffect(() => {
+    let getColorsEndpoint = backendUrl + "/get-colors";
+    fetch(getColorsEndpoint, { mode: "cors" })
+      .then((response) => {
+        return response;
+      })
+      .then((data) => {
+        console.log(data.body)
+        let colorData = new Uint8Array(data, 0, data.byteLength);
+        console.log(colorData);
+
+        // Convert ArrayBuffer to base-10 array
+        // const base10Array = [];
+        // const view = new Uint8Array(data);
+        // for (let i = 0; i < view.length; i++) {
+        //   base10Array.push(view[i]);
+        // }
+
+        // setColors(base10Array);
+        // console.log(base10Array)
+      })
+      .catch((error) => {
+        console.error(error);
```

Comment:
Can you do `setColors(canvasConfig.colors)` here if there's an error?

Reply:
Done.